### PR TITLE
Fix convert

### PR DESCRIPTION
**Pull request by mohamed82008**
This PR fixes some `convert` bugs.

### CODE REVIEW

**Comment by devmotion** [comment]
It would be good to add some tests for these methods.

**Comment by mohamed82008** [comment]
Done.

File: src/tarray.jl [hunk 1]
```diff
@@ -87,14 +87,20 @@ function Base.pop!(S::TArray)
     pop!(d)
 end
 
+function Base.convert(::Type{TArray{T, N}}, x::Array{T, N}) where {T, N}
+    return convert(TArray, x)
+end
 function Base.convert(::Type{TArray}, x::Array)
```
Comment (devmotion) [inline src/tarray.jl:93]: Wouldn't it be more natural to define
```suggestion
function Base.convert(::Type{TArray}, x::Array)
    return convert(TArray{eltype(x),ndims(x)}, x)
end
function Base.convert(::Type{TArray{T,N}}, x::Array{T,N}) where {T,N}
```
if `convert(::Type{TArray{T,N}}, ::Array{T,N})` is needed?

Comment (mohamed82008) [reply src/tarray.jl:93]: Sure.

File: src/tarray.jl [hunk 2]
```diff
@@ -88,15 +88,21 @@ function Base.pop!(S::TArray)
 end
 
 function Base.convert(::Type{TArray}, x::Array)
-    res = TArray{typeof(x[1]),ndims(x)}();
+    return convert(TArray{eltype(x),ndims(x)}, x)
+end
+function Base.convert(::Type{TArray{T,N}}, x::Array{T,N}) where {T,N}
+    res = TArray{T,N}();
```
Comment (devmotion) [inline src/tarray.jl:94]:
```suggestion
    res = TArray{T,N}()
```

**Pull request merged**